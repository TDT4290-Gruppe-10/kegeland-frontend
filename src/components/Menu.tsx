import { VStack, Text, Divider, HStack, Icon, Box } from '@chakra-ui/react';
import { ComponentProps } from 'react';
import { IconType } from 'react-icons/lib';
import { Link, To } from 'react-router-dom';

type MenuItemProps = {
  title: string;
  to: To;
  icon?: IconType;
};

const Item: React.FC<MenuItemProps> = ({ title, to, icon }) => {
  return (
    <HStack
      spacing={1}
      marginBottom={2}
      width="full"
      _hover={{ color: 'primary.600', cursor: 'pointer' }}
      transitionDuration="300ms">
      {icon && <Icon as={icon} color="primary.600" fontSize={24} />}
      <Link to={to}>
        <Text fontWeight="semibold">{title}</Text>
      </Link>
    </HStack>
  );
};

type MenuComponents = {
  Item: typeof Item;
};

type MenuProps = {
  title: string;
  head?: React.ReactNode;
  children?: React.ReactNode;
} & ComponentProps<typeof VStack>;

const Menu: React.FunctionComponent<MenuProps> & MenuComponents = ({
  title,
  head,
  children,
  ...props
}) => {
  return (
    <VStack
      {...props}
      width="full"
      spacing={2}
      paddingTop={4}
      alignItems="flex-start">
      {head && <Box paddingBottom={2}>{head}</Box>}
      {head && <Divider />}
      <Text
        color="primary.600"
        fontWeight="semibold"
        width="full"
        textTransform="uppercase">
        {title}
      </Text>
      {children}
      <Divider />
    </VStack>
  );
};

Menu.Item = Item;

export default Menu;
