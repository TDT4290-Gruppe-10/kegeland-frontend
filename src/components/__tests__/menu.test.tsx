import renderer from 'react-test-renderer';

import Menu from '../Menu';

describe('Test manu', () => {
  it('renders correctlyu', () => {
    const tree = renderer.create(<Menu title={'hello'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
