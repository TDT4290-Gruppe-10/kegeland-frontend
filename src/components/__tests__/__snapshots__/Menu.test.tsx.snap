// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test manu renders correctlyu 1`] = `
<div
  className="chakra-stack css-7nrznn"
>
  <p
    className="chakra-text css-199kdpe"
  >
    hello
  </p>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1x4iw6"
  />
</div>
`;
